currency/manager.go

```diff
@@ -435,34 +431,27 @@ func (p *PairsManager) SetAssetEnabled(a asset.Item, enabled bool) error {
 }
 
 // Load sets the pair manager from a seed without copying mutexes
-func (p *PairsManager) Load(seed *PairsManager) error {
-	if seed == nil {
-		return fmt.Errorf("%w PairsManager", common.ErrNilPointer)
-	}
+func (p *PairsManager) Load(seed *PairsManager) {
 	p.mutex.Lock()
 	defer p.mutex.Unlock()
 	seed.mutex.RLock()
 	defer seed.mutex.RUnlock()
 
-	var pN PairsManager
-	j, err := json.Marshal(seed)
-	if err != nil {
-		return err
-	}
-	err = json.Unmarshal(j, &pN)
-	if err != nil {
-		return err
-	}
-	p.BypassConfigFormatUpgrades = pN.BypassConfigFormatUpgrades
-	if pN.UseGlobalFormat {
-		p.UseGlobalFormat = pN.UseGlobalFormat
-		p.RequestFormat = pN.RequestFormat
-		p.ConfigFormat = pN.ConfigFormat
+	p.BypassConfigFormatUpgrades = seed.BypassConfigFormatUpgrades
+	p.UseGlobalFormat = seed.UseGlobalFormat
+	p.LastUpdated = seed.LastUpdated
+	p.Pairs = seed.Pairs.clone()
+	p.RequestFormat = seed.RequestFormat.clone()
+	p.ConfigFormat = seed.ConfigFormat.clone()
+	p.matcher = map[key]*Pair{}
+
+	// Re-index matcher
+	for a, fs := range p.Pairs {
```

Comment:
Suggestion: This is done a few times could make a method for it.

Reply:
This exposed that the matcher index was never flushed when storing pairs.
We have three options:
1. Blow away the whole index each time regardless of asset updated
2. Restructure the key to make it fast to blow away just one asset.
3. Walk the current map and delete by asset first

Since store pair is relatively infrequent, I've opted to blow away the whole index each time, for simplicity.

Fixed 9005629

Reply:
🔥 Nice 👍